lib/pdn/pdn.c

```diff
@@ -196,18 +197,11 @@ int pdn_ctx_create(uint8_t *cid, pdn_event_handler_t cb)
 		return -ENOMEM;
 	}
 
-	err = at_cmd_write("AT%XNEWCID?", at_buf, sizeof(at_buf), NULL);
-	if (err) {
+	err = nrf_modem_at_scanf("AT%XNEWCID?", "%%XNEWCID: %d", &pdn->context_id);
+	if (err < 0) {
```

Comment:
Should 0 be also handled as an error?

Reply:
Thanks, no this API behaves slightly differently from `scanf` insofar that it returns a negative errno if there are no matches (instead of returning Zero).